File: python/problem057.py
denominator = 1
numerator = 1

index = 0
result = 0
while index <= 1000:
    index += 1
    temp_denominator = numerator + denominator
    temp_numerator = numerator + 2 * denominator
    denominator = temp_denominator
    numerator = temp_numerator
    d_str = str(denominator)
    n_str = str(numerator)
    if len(n_str) > len(d_str):
        result += 1

print(result)

File: java/src/Problem057.java
import java.math.BigInteger;

public class Problem057 {

    int result(){
        BigInteger denominator = new BigInteger("1");
        BigInteger numerator = new BigInteger("1");
        int index = 0;
        int value = 0;
        while(index <= 1000){
            index = index + 1;
            BigInteger tempDenominator = numerator.add(denominator);
            BigInteger tempNumerator = numerator.add(denominator.multiply(new BigInteger("2")));
            denominator = tempDenominator;
            numerator = tempNumerator;
            String denomiatorString = denominator.toString();
            String numeratorString = numerator.toString();
            if(numeratorString.length() > denomiatorString.length()){
                value = value + 1;
            }
        }
        return value;
    }

    public static void main(String args[]){
        Problem057 problem = new Problem057();
        System.out.println(problem.result());
    }
}
